### ISSUE TITLE
code-server 3.2.0 starting but can't access the website

### ISSUE THREAD

**Issue by MrHelloSir** [cdr/code-server#1800]
I receive the following response when running `code-server` on my ubuntu 18.04 VPS:
```
info  code-server 3.2.0 fd36a99a4c78669970ebc4eb05768293b657716f
info  HTTP server listening on http://127.0.0.1:8080
info    - Password is `somepassword`
info      - To use your own password set the PASSWORD environment variable
info      - To disable use `--auth none`
info    - Not serving HTTPS
info  Automatic updates are enabled
```
But I can't access the website. (MY VPS IP + port 8080 which returns This site can’t be reached)
- Web Browser: Chrome
- `code-server --version`: 3.2.0 fd36a99a4c78669970ebc4eb05768293b657716f

**Comment by nhooyr** [comment]
It's listening on `http://127.0.0.1:8080`.

Pass `--host 0.0.0.0` to expose it.

Also, please upgrade to the latest version and give https://github.com/cdr/code-server/blob/master/doc/guide.md a read.

**Comment by MrHelloSir** [comment]
Thanks for your help. I've upgraded to
`3.4.1 48f7c2724827e526eeaa6c2c151c520f48a61259`

Now I run code-server --port 8446 --host 0.0.0.0 and I still can't access IP:8446

**Comment by MrHelloSir** [comment]
![Screenshot_31](https://user-images.githubusercontent.com/63464262/84497156-e25d1d80-acb6-11ea-806f-2a92a85245e6.png)
Here you can see what happens. Tried to remove and reinstall it. I am allowing 8446 on my UFW firewall (also tried with 8443 but there shall be no change)

**Comment by MrHelloSir** [comment]
@nhooyr Could you assist? Please.

**Comment by nhooyr** [comment]
You're not running it with `--host`, it says it's still listening on `127.0.0.1`.

**Comment by nhooyr** [comment]
Also, please file future questions at https://www.reddit.com/r/codeserver!

**Comment by MrHelloSir** [comment]
> You're not running it with `--host`, it says it's still listening on `127.0.0.1`.

Even if I use host it still won't work.

**Comment by madcpt** [comment]
Are you hosting it on `0.0.0.0` (by using `--host 0.0.0.0`)?
Because `127.0.0.1` won't expose the service!

**Comment by nicolharico** [comment]
Hi,
Sorry to comment of this closed thread, but I’m facing somehow the same issue.

I specifically pass the flag —host 0.0.0.0 to the code-server in the service, but it always starts the server listening on local host 127.0.0.1:8080, making it natively not reachable from external devices. 

My code server config file contains the bind address 0.0.0.0 and, when I simply start the code-server binary without any flag, it smoothly starts on 0.0.0.0. When executing code-server binary from a service, wether try to force the host or not, it stays on 127.0.0.1 :(

I’ll can post my code-server config file, the code-server service, the output I got from executing code-server and from starting systemd code-server service if needed. 

Thanks

**Comment by code-asher** [comment]
@nicolharico I would recommend a separate issue!

But to kick things off is your service being started with a different user? The config file is read from the user's home directory.

**Comment by code-asher** [comment]
Oh I see you've tried passing `--host` to the service so that can't be it. Please do open a new issue with your config file and output with `--debug`. Thanks!

**Comment by nicolharico** [comment]
Hi ! Sorry for my response being late, l've been out of office lately.

Sure I will open an issue with my detailed configs, thanks for your support and reactivity :)

**Comment by caochitam** [comment]
Simple solution for me is update config.yaml and restart code-server
`bind-addr: 0.0.0.0:8080`
![image](https://user-images.githubusercontent.com/5225981/206316630-e1458845-7d21-4f78-bc84-53e5215dd71e.png)
Ofcouse, i sure 8080 port is can access from internet by using a simple http-server

**Comment by jon-peel** [comment]
Hi, I am having the same problem, has anyone figured it out.
@n00dleee, did you open a new issue, can I get a link if there is more info there.

I have tried setting the address in the config and with the `--host` argument, but I am not able to connect from my computer.
I have tried setting the address to four 0s, and I have also tried the computers IP.
Nothing lets me connect.

My config is
```yaml
bind-addr: 0.0.0.0:8080
auth: password
password: 1234
cert: false
```
